Unable to compile under Ubuntu 14.04 (ffmpeg)

Starting with commit 504b771, compilation under ubuntu 14.04 fails with:
```
/home/giovanni/remmina_devel/FreeRDP/libfreerdp/codec/h264_ffmpeg.c: In function ‘libavcodec_destroy_encoder’:
/home/giovanni/remmina_devel/FreeRDP/libfreerdp/codec/h264_ffmpeg.c:62:3: warning: implicit declaration of function ‘av_free’ [-Wimplicit-function-declaration]
   av_free(sys->codecEncoderContext);
   ^
/home/giovanni/remmina_devel/FreeRDP/libfreerdp/codec/h264_ffmpeg.c: In function ‘libavcodec_create_encoder’:
/home/giovanni/remmina_devel/FreeRDP/libfreerdp/codec/h264_ffmpeg.c:119:26: error: ‘AVCodecContext’ has no member named ‘framerate’
  sys->codecEncoderContext->framerate = (AVRational)
                          ^
/home/giovanni/remmina_devel/FreeRDP/libfreerdp/codec/h264_ffmpeg.c:130:2: warning: implicit declaration of function ‘av_opt_set’ [-Wimplicit-function-declaration]
  av_opt_set(sys->codecEncoderContext->priv_data, "preset", "fast", 0);
  ^
/home/giovanni/remmina_devel/FreeRDP/libfreerdp/codec/h264_ffmpeg.c: In function ‘libavcodec_compress’:
/home/giovanni/remmina_devel/FreeRDP/libfreerdp/codec/h264_ffmpeg.c:224:17: error: ‘AVFrame’ has no member named ‘colorspace’
  sys->videoFrame->colorspace = AVCOL_SPC_BT709;
                 ^
/home/giovanni/remmina_devel/FreeRDP/libfreerdp/codec/h264_ffmpeg.c:269:3: warning: implicit declaration of function ‘av_err2str’ [-Wimplicit-function-declaration]
   WLog_ERR(TAG, "Failed to encode video frame (%s [%d])",
   ^
make[2]: *** [libfreerdp/CMakeFiles/freerdp.dir/codec/h264_ffmpeg.c.o] Error 1
make[1]: *** [libfreerdp/CMakeFiles/freerdp.dir/all] Error 2
make: *** [all] Error 2
```

Compiling against libavutil-dev  6:9.20-0ubuntu amd64

I don't think this issue it's very important: we can stick to the previous working FreeRDP commit for our Ubuntu 14.04 packaging. Just let us know if you plan to fix backward compatibility or not.
Cc: @antenore and @dfiloni

--- a/libfreerdp/codec/h264_ffmpeg.c
+++ b/libfreerdp/codec/h264_ffmpeg.c
@@ -24,12 +24,26 @@
 #include <libavcodec/avcodec.h>
 #include <libavutil/avutil.h>
 #include <libavutil/opt.h>
+#include <libavutil/mem.h>
 
 #define TAG FREERDP_TAG("codec")
 
 /* Fallback support for older libavcodec versions */
 #if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(54, 59, 100)
 #define AV_CODEC_ID_H264 CODEC_ID_H264
+#endif
+
+/* Ubuntu 14.04 ships without the functions provided by avutil,
+ * so define error to string methods here. */
+#if !defined(av_err2str)
+static inline char* error_string(char* errbuf, size_t errbuf_size, int errnum)
+{
+	av_strerror(errnum, errbuf, errbuf_size);
+	return errbuf;
+}
+
+#define av_err2str(errnum) \
+	error_string((char[64]){0}, 64, errnum)
 #endif
 
 struct _H264_CONTEXT_LIBAVCODEC
@@ -116,10 +130,12 @@ static BOOL libavcodec_create_encoder(H264_CONTEXT* h264)
 	sys->codecEncoderContext->width = h264->width;
 	sys->codecEncoderContext->height = h264->height;
 	sys->codecEncoderContext->delay = 0;
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(56, 13, 100)
 	sys->codecEncoderContext->framerate = (AVRational)
 	{
 		h264->FrameRate, 1
 	};
+#endif
 	sys->codecEncoderContext->time_base = (AVRational)
 	{
 		1, h264->FrameRate
@@ -127,6 +143,7 @@ static BOOL libavcodec_create_encoder(H264_CONTEXT* h264)
 	sys->codecEncoderContext->gop_size = 0;
 	sys->codecEncoderContext->max_b_frames = 0;
 	sys->codecEncoderContext->pix_fmt = AV_PIX_FMT_YUV420P;
+	sys->codecEncoderContext->colorspace = AVCOL_SPC_BT709;
 	av_opt_set(sys->codecEncoderContext->priv_data, "preset", "fast", 0);
 	av_opt_set(sys->codecEncoderContext->priv_data, "vprofile", "baseline", 0);
 	av_opt_set(sys->codecEncoderContext->priv_data, "tune", "zerolatency", 0);
@@ -221,7 +238,9 @@ static int libavcodec_compress(H264_CONTEXT* h264, BYTE** ppDstData, UINT32* pDs
 	sys->videoFrame->format = sys->codecEncoderContext->pix_fmt;
 	sys->videoFrame->width  = sys->codecEncoderContext->width;
 	sys->videoFrame->height = sys->codecEncoderContext->height;
+#if LIBAVUTIL_VERSION_INT >= AV_VERSION_INT(52, 48, 100)
 	sys->videoFrame->colorspace = AVCOL_SPC_BT709;
+#endif
 	sys->videoFrame->data[0] = h264->pYUVData[0];
 	sys->videoFrame->data[1] = h264->pYUVData[1];
 	sys->videoFrame->data[2] = h264->pYUVData[2];
@@ -310,7 +329,6 @@ static void libavcodec_uninit(H264_CONTEXT* h264)
 	}
 
 	libavcodec_destroy_encoder(h264);
-
 	free(sys);
 	h264->pSystemData = NULL;
 }
@@ -379,7 +397,6 @@ static BOOL libavcodec_init(H264_CONTEXT* h264)
 	}
 
 	sys->videoFrame->pts = 0;
-
 	return TRUE;
 EXCEPTION:
 	libavcodec_uninit(h264);

--- a/libfreerdp/core/nla.c
+++ b/libfreerdp/core/nla.c
@@ -1668,7 +1668,7 @@ int nla_recv_pdu(rdpNla* nla, wStream* s)
 
 		    default:
 			    WLog_ERR(TAG, "SPNEGO failed with NTSTATUS: 0x%08"PRIX32"", nla->errorCode);
-				code = FREERDP_ERROR_CONNECT_FAILED;
+				code = FREERDP_ERROR_AUTHENTICATION_FAILED;
 			    break;
 		}
 		freerdp_set_last_error(nla->instance->context, code);

